subsys/net/lib/http/http_server_core.c

```diff
@@ -873,7 +873,21 @@ int http_server_sendall(struct http_client_ctx *client, const void *buf, size_t
 	while (len) {
 		ssize_t out_len = zsock_send(client->fd, buf, len, 0);
 
-		if (out_len < 0) {
+		if ((out_len == 0) || (out_len < 0 && errno == EAGAIN)) {
+			struct zsock_pollfd pfd;
+			int pollres;
+
+			pfd.fd = sock;
+			pfd.events = ZSOCK_POLLOUT;
+			pollres = zsock_poll(&pfd, 1, 0);
+			if (pollres == 0) {
```

Comment:
Wouldn't we end up in `ETIMEDOUT` error now in case when the socket would normally block (given the `zsock_poll()` has 0 timeout)?

Reply:
You are right, that's why I asked about 0 value. Basically with 0 this PR doesn't really change previous behavior, so we need some meaningful value.